### PR TITLE
Add test to zero decrement

### PR DESCRIPTION
**Pull request by Pythonlover-1**
Не было проверки dec от 0 (и соответственно от big_integer, созданного дефолтным конструктором). Если делать dec через вычитание маленького из большого и не учитывать, что может произойти переход из 0 в отрицательные числа, тесты принимали такие результаты при dec от 0, как 0 и 4294967295.
Аналогичные проблемы были с inc от -1

### CODE REVIEW

File: tests.cpp [hunk 1]
```diff
@@ -330,6 +330,21 @@ TEST(correctness, decrement) {
 
   EXPECT_EQ(41, pre);
   EXPECT_EQ(41, post);
+  // Не было проверки dec от 0 (и соответственно от big_integer,
+  // созданного дефолтным конструктором)
+  // Если делать dec через вычитание маленького из большого и не учитывать, что может
+  // произойти переход из 0 в отриц числа, тесты принимали такие результаты при dec от 0, как
+  // 0 и 4294967295
```
Comment (dedlocc) [inline tests.cpp:337]: это комментарий к PR, а не к коду, зачем он здесь?

File: tests.cpp [hunk 2]
```diff
@@ -330,6 +330,21 @@ TEST(correctness, decrement) {
...
+  big_integer b;
+  big_integer pre_b = --b;
+  big_integer post_b = b--;
+  EXPECT_EQ(-1, pre_b);
+  EXPECT_EQ(-1, post_b);
+  big_integer c = 0;
+  big_integer pre_c = --c;
+  big_integer post_c = c--;
+  EXPECT_EQ(-1, pre_c);
+  EXPECT_EQ(-1, post_c);
```
Comment (dedlocc) [inline tests.cpp:347]: разве существует адекватная реализация бигинта, в котором `b` и `c` не эквивалентны? ну и в любом случае проверка их эквивалентности была бы отдельным тестом

File: tests.cpp [hunk 3]
```diff
@@ -330,6 +330,21 @@ TEST(correctness, decrement) {
...
+  // Не было проверки dec от 0 (и соответственно от big_integer,
+  // созданного дефолтным конструктором)
+  // Если делать dec через вычитание маленького из большого и не учитывать, что может
+  // произойти переход из 0 в отриц числа, тесты принимали такие результаты при dec от 0, как
+  // 0 и 4294967295
+  big_integer b;
+  big_integer pre_b = --b;
+  big_integer post_b = b--;
+  EXPECT_EQ(-1, pre_b);
+  EXPECT_EQ(-1, post_b);
```
Comment (dedlocc) [inline tests.cpp:342]: наверное хочется тогда аналогичный тест и для `++`?

**Review by dedlocc — CHANGES_REQUESTED** [review]
давай сделаем это отдельным тестом/тестами, а не пихать в существующие

File: tests.cpp (hunk 3 above)
Comment (Pythonlover-1) [reply tests.cpp:342]: Он сделан в следующем pr)

Comment (Pythonlover-1) [reply tests.cpp:344]: fixed

File: tests.cpp (hunk 1 above)
Comment (Pythonlover-1) [reply tests.cpp]: fixed

File: tests.cpp (hunk 2 above)
Comment (Pythonlover-1) [reply tests.cpp]: fixed

**Review by dedlocc — CHANGES_REQUESTED** [review]
раздели на 2 теста

**Review by dedlocc — APPROVED** [review]
